File: problemList/slidingWIndows/solution/LC1208.java
package problemList.slidingWIndows.solution;

public class LC1208 {
    public int equalSubstring(String s, String t, int maxCost) {
        int n = s.length(), left = 0, ret = 0, curCost = 0;
        for(int i = 0;i < n;i++){
            curCost += (int)Math.abs(s.charAt(i) - t.charAt(i));
            while(curCost > maxCost){
                curCost -= (int)Math.abs(s.charAt(left) - t.charAt(left));
                left++;
            }
            ret = Math.max(ret, i - left + 1);
        }
        return ret;
    }
}

File: SolutionsWithCorCpp/problem_list_0x3f/sliding_window/solution/LC1208.cpp
#include <algorithm>
#include <iostream>
#include <unordered_map>
#include <vector>
using namespace std;

int equalSubstring(string s, string t, int maxCost) {
	int n = s.size(), curCost = 0, r = 0, ret = 0;
	for (int i = 0; r < n; r++) {
		curCost += abs(s[r] - t[r]);
		while (curCost > maxCost && i < n) {
			curCost -= abs(s[i] - t[i]);
			i++;
		}
		if (i <= r) ret = max(ret, r - i + 1);
	}
	return ret;
}

int main() {
	string s = "abcd";
	string t = "cdef";
	int maxCost = 3;
	cout << equalSubstring(s, t, maxCost) << endl;

	return 0;
}